packages/camera/camera/android/src/main/java/io/flutter/plugins/camera/Camera.java

```diff
@@ -470,9 +470,7 @@ public void onListen(Object o, EventChannel.EventSink imageStreamSink) {
           }
 
           @Override
-          public void onCancel(Object o) {
-            imageStreamReader.setOnImageAvailableListener(null, null);
```

Comment:
Why we should remove this ?

`imageStreamReader.setOnImageAvailableListener(null, null);`

Reply:
Because we call `setOnImageAvailableListener` in `stopImageStream` this would be redundant.

Reply:
i dont think so , because `stopImageStream`  is in `MethodChannel` and you removed `onCancel` from `StreamHandler`

Reply:
The thing here is that the `CameraController.stopImageStream` executes both the `stopImageStream` method on the `MethodChannel` and will cancel the stream on the `EventChannel`. Making the logic in the `onCancel` method redundant:

```dart
  Future<void> stopImageStream() async {
    assert(defaultTargetPlatform == TargetPlatform.android ||
        defaultTargetPlatform == TargetPlatform.iOS);

   // Left out error handling to keep example compact.

    try {
      value = value.copyWith(isStreamingImages: false);
      await _channel.invokeMethod<void>('stopImageStream');
    } on PlatformException catch (e) {
      throw CameraException(e.code, e.message);
    }

    await _imageStreamSubscription.cancel();
    _imageStreamSubscription = null;
  }
```